### PR TITLE
[WIP] Adds simple control flag

### PR DESCRIPTION
**Pull request by chaosvolt**
First attempt at adding a vehicle part flag that allows controlling a vehicle and honking horns (will be used for bicycles), but with no other advanced functions.
- Will implement it for certain parts later, assuming it works.

### CODE REVIEW

**Comment by chaosvolt** [comment]
It was inevitable.

**Comment by chaosvolt** [comment]
Hoooly shit. Have I hit my record for most failed commits in a row yet? -_-

And now to actually implement the bloody vehicle part flag and try to figure out how to get my compiler working again to test whether it even works.

File: src/vehicle.cpp [hunk 1]
```diff
@@ -109,6 +109,15 @@ enum vehicle_controls {
  toggle_scoop
 };
 
+enum vehicle_controls_simple {
+ toggle_cruise_control_simple,
```
Comment (Coolthulhu) [inline src/vehicle.cpp]: 4 spaces

File: src/vehicle.cpp [hunk 2]
```diff
@@ -109,6 +109,15 @@ enum vehicle_controls {
  toggle_scoop
 };
 
+enum vehicle_controls_simple {
```
Comment (Coolthulhu) [inline src/vehicle.cpp]: No need to make a separate enum for all this stuff, you could (probably should) reuse the enum that already exists.
You could make a set of control options available for simple controls like this:
```
const auto simple_options = std::set<vehicle_controls>() { toggle_cruise_control, activate_horn, [rest of the stuff here], convert_vehicle };
```
And check if a given control option is available at simple controls with `simple_options.count( option ) > 0`.

File: src/vehicle.cpp [hunk 3]
```diff
@@ -1296,6 +1309,98 @@ void vehicle::use_controls()
     refresh();
 }
 
+void vehicle::use_controls_simple()
```
Comment (Coolthulhu) [inline src/vehicle.cpp]: A separate function that implements part of the functionality of a bigger one is a pain to maintain.

You could split the `use_controls` section into a simple part and advanced part. Then you could just have the advanced part in an `if( not_simple )`. You wouldn't need to handle advanced options in a switch, just populating the menu.

**Comment by kevingranade** [comment]
Outside of occasional lapses, you shouldn't be pushing commits that don't compile in the first place, if your build setup isn't working, get that fixed first before pushing more commits, the build bot is not there for you to test if your commits compile.
More generally, you should be building and testing your code before pushing, unless you need feedback on something specific you don't understand.

**Comment by chaosvolt** [comment]
Ack. Sorry. I've still yet to get the compiler to actually WORK though. 90% of the time I can test changes in an earlier build, but that's only doable with JSON-only change, not anything source-related.

And...hmm. Would need to work out how to change that to allow that distinction between simple and complex functions. @_@

**Comment by chaosvolt** [comment]
Augh. Well, while I continue to wrestle with everything the compiler needs to not puke up fatal errors...at this point I'm halfway to scrapping this PR. ._.

**Comment by chaosvolt** [comment]
And MinGW's 1000-year reign of compiling ends with it reminding me that I forgot to give it the freetype stuff. Fuck me. ._.

**Comment by Coolthulhu** [comment]
Don't compile with tiles or localization until you're sure you can compile anything.

**Comment by chaosvolt** [comment]
I was already setting it to "make TILES=0 NATIVE=win32 LOCALIZE=0" though. >_>

I'm guessing I should've omitted those entries entirely instead?

**Comment by Coolthulhu** [comment]
Makefile is kinda weirdly about those: you want `LOCALIZE=0`, but you don't want any mention of `TILES` or it will think you want them.

So,
```
make NATIVE=win32 LOCALIZE=0
```

You can add `RELEASE=1` for faster linking and execution if you don't intend to debug it with a debugger.

**Comment by chaosvolt** [comment]
Hmm. Let me try that then.

Still, it's getting to the point where I'm about ready to just say "fuck it" and leave this isdea to someone else. I still haven't even begun to see if the current lazy method of adding a simple control method works as intended, let alone changed it to the way you suggested I do. >.<

**Comment by kevingranade** [comment]
Just FYI, release is usually slower to compile, not faster.  Gcc does some
heuristic-search type things when any optimization is turned on.

**Comment by chaosvolt** [comment]
Weird, given the different between trying it with and without the release option.

Right now I'm not even sure what my next step is then. ._.

**Comment by chaosvolt** [comment]
Hmm. Still unsure if either way of adding simple options can be rigged up properly to associate the simple options with the relevant vehicle part flag.

**Comment by chaosvolt** [comment]
Okay, I'm...still stumped here. Just...fuck it. I give up.

**Comment by illi-kun** [comment]
I can try to implement that functionality.

**Pull request closed without merging**